// @serverless/sdk-schema/dist/tags/aws.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Long = require('long');
var _m0 = require('protobufjs/minimal');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var Long__default = /*#__PURE__*/_interopDefaultLegacy(Long);
var _m0__default = /*#__PURE__*/_interopDefaultLegacy(_m0);

/* eslint-disable */
function createBaseHttpTags() {
    return {
        method: "",
        protocol: "",
        host: "",
        path: "",
        queryParameterNames: [],
        requestHeaderNames: [],
        statusCode: undefined,
        errorCode: undefined,
    };
}
const HttpTags = {
    encode(message, writer = _m0__default["default"].Writer.create()) {
        if (message.method !== "") {
            writer.uint32(10).string(message.method);
        }
        if (message.protocol !== "") {
            writer.uint32(18).string(message.protocol);
        }
        if (message.host !== "") {
            writer.uint32(26).string(message.host);
        }
        if (message.path !== "") {
            writer.uint32(34).string(message.path);
        }
        for (const v of message.queryParameterNames) {
            writer.uint32(50).string(v);
        }
        for (const v of message.requestHeaderNames) {
            writer.uint32(66).string(v);
        }
        if (message.statusCode !== undefined) {
            writer.uint32(72).uint32(message.statusCode);
        }
        if (message.errorCode !== undefined) {
            writer.uint32(98).string(message.errorCode);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0__default["default"].Reader ? input : _m0__default["default"].Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseHttpTags();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.method = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.protocol = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.host = reader.string();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.path = reader.string();
                    continue;
                case 6:
                    if (tag !== 50) {
                        break;
                    }
                    message.queryParameterNames.push(reader.string());
                    continue;
                case 8:
                    if (tag !== 66) {
                        break;
                    }
                    message.requestHeaderNames.push(reader.string());
                    continue;
                case 9:
                    if (tag !== 72) {
                        break;
                    }
                    message.statusCode = reader.uint32();
                    continue;
                case 12:
                    if (tag !== 98) {
                        break;
                    }
                    message.errorCode = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            method: isSet$1(object.method) ? String(object.method) : "",
            protocol: isSet$1(object.protocol) ? String(object.protocol) : "",
            host: isSet$1(object.host) ? String(object.host) : "",
            path: isSet$1(object.path) ? String(object.path) : "",
            queryParameterNames: Array.isArray(object === null || object === void 0 ? void 0 : object.queryParameterNames)
                ? object.queryParameterNames.map((e) => String(e))
                : [],
            requestHeaderNames: Array.isArray(object === null || object === void 0 ? void 0 : object.requestHeaderNames)
                ? object.requestHeaderNames.map((e) => String(e))
                : [],
            statusCode: isSet$1(object.statusCode) ? Number(object.statusCode) : undefined,
            errorCode: isSet$1(object.errorCode) ? String(object.errorCode) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        message.method !== undefined && (obj.method = message.method);
        message.protocol !== undefined && (obj.protocol = message.protocol);
        message.host !== undefined && (obj.host = message.host);
        message.path !== undefined && (obj.path = message.path);
        if (message.queryParameterNames) {
            obj.queryParameterNames = message.queryParameterNames.map((e) => e);
        }
        else {
            obj.queryParameterNames = [];
        }
        if (message.requestHeaderNames) {
            obj.requestHeaderNames = message.requestHeaderNames.map((e) => e);
        }
        else {
            obj.requestHeaderNames = [];
        }
        message.statusCode !== undefined && (obj.statusCode = Math.round(message.statusCode));
        message.errorCode !== undefined && (obj.errorCode = message.errorCode);
        return obj;
    },
    create(base) {
        return HttpTags.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c, _d, _e, _f, _g, _h;
        const message = createBaseHttpTags();
        message.method = (_a = object.method) !== null && _a !== void 0 ? _a : "";
        message.protocol = (_b = object.protocol) !== null && _b !== void 0 ? _b : "";
        message.host = (_c = object.host) !== null && _c !== void 0 ? _c : "";
        message.path = (_d = object.path) !== null && _d !== void 0 ? _d : "";
        message.queryParameterNames = ((_e = object.queryParameterNames) === null || _e === void 0 ? void 0 : _e.map((e) => e)) || [];
        message.requestHeaderNames = ((_f = object.requestHeaderNames) === null || _f === void 0 ? void 0 : _f.map((e) => e)) || [];
        message.statusCode = (_g = object.statusCode) !== null && _g !== void 0 ? _g : undefined;
        message.errorCode = (_h = object.errorCode) !== null && _h !== void 0 ? _h : undefined;
        return message;
    },
};
if (_m0__default["default"].util.Long !== Long__default["default"]) {
    _m0__default["default"].util.Long = Long__default["default"];
    _m0__default["default"].configure();
}
function isSet$1(value) {
    return value !== null && value !== undefined;
}

/* eslint-disable */
const protobufPackage = "serverless.instrumentation.tags.v1";
/**
 * A Lambda function invocation can have one of the following
 * outcomes upon completion.
 */
exports.AwsLambdaTags_Outcome = void 0;
(function (AwsLambdaTags_Outcome) {
    /** OUTCOME_UNSPECIFIED - No outcome was registered. Either information on the outcome was not disclosed (which should never be the case), or the function invocation has not been finalized yet */
    AwsLambdaTags_Outcome[AwsLambdaTags_Outcome["OUTCOME_UNSPECIFIED"] = 0] = "OUTCOME_UNSPECIFIED";
    /** OUTCOME_SUCCESS - Function handler returned successfully */
    AwsLambdaTags_Outcome[AwsLambdaTags_Outcome["OUTCOME_SUCCESS"] = 1] = "OUTCOME_SUCCESS";
    /** OUTCOME_ERROR_INITIALIZATION - Function crashed at initialization */
    AwsLambdaTags_Outcome[AwsLambdaTags_Outcome["OUTCOME_ERROR_INITIALIZATION"] = 2] = "OUTCOME_ERROR_INITIALIZATION";
    /** OUTCOME_ERROR_UNHANDLED - Function crashed in an unhandled way at invocation (a result of either uncaught exception or unhandled rejection) */
    AwsLambdaTags_Outcome[AwsLambdaTags_Outcome["OUTCOME_ERROR_UNHANDLED"] = 3] = "OUTCOME_ERROR_UNHANDLED";
    /** OUTCOME_ERROR_TIMEOUT - Function timed out */
    AwsLambdaTags_Outcome[AwsLambdaTags_Outcome["OUTCOME_ERROR_TIMEOUT"] = 4] = "OUTCOME_ERROR_TIMEOUT";
    /** OUTCOME_ERROR_HANDLED - Function handler resolved with an error (either error was passed to lambda callback, or async handler resolved with error rejection) */
    AwsLambdaTags_Outcome[AwsLambdaTags_Outcome["OUTCOME_ERROR_HANDLED"] = 5] = "OUTCOME_ERROR_HANDLED";
    AwsLambdaTags_Outcome[AwsLambdaTags_Outcome["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(exports.AwsLambdaTags_Outcome || (exports.AwsLambdaTags_Outcome = {}));
function awsLambdaTags_OutcomeFromJSON(object) {
    switch (object) {
        case 0:
        case "OUTCOME_UNSPECIFIED":
            return exports.AwsLambdaTags_Outcome.OUTCOME_UNSPECIFIED;
        case 1:
        case "OUTCOME_SUCCESS":
            return exports.AwsLambdaTags_Outcome.OUTCOME_SUCCESS;
        case 2:
        case "OUTCOME_ERROR_INITIALIZATION":
            return exports.AwsLambdaTags_Outcome.OUTCOME_ERROR_INITIALIZATION;
        case 3:
        case "OUTCOME_ERROR_UNHANDLED":
            return exports.AwsLambdaTags_Outcome.OUTCOME_ERROR_UNHANDLED;
        case 4:
        case "OUTCOME_ERROR_TIMEOUT":
            return exports.AwsLambdaTags_Outcome.OUTCOME_ERROR_TIMEOUT;
        case 5:
        case "OUTCOME_ERROR_HANDLED":
            return exports.AwsLambdaTags_Outcome.OUTCOME_ERROR_HANDLED;
        case -1:
        case "UNRECOGNIZED":
        default:
            return exports.AwsLambdaTags_Outcome.UNRECOGNIZED;
    }
}
function awsLambdaTags_OutcomeToJSON(object) {
    switch (object) {
        case exports.AwsLambdaTags_Outcome.OUTCOME_UNSPECIFIED:
            return "OUTCOME_UNSPECIFIED";
        case exports.AwsLambdaTags_Outcome.OUTCOME_SUCCESS:
            return "OUTCOME_SUCCESS";
        case exports.AwsLambdaTags_Outcome.OUTCOME_ERROR_INITIALIZATION:
            return "OUTCOME_ERROR_INITIALIZATION";
        case exports.AwsLambdaTags_Outcome.OUTCOME_ERROR_UNHANDLED:
            return "OUTCOME_ERROR_UNHANDLED";
        case exports.AwsLambdaTags_Outcome.OUTCOME_ERROR_TIMEOUT:
            return "OUTCOME_ERROR_TIMEOUT";
        case exports.AwsLambdaTags_Outcome.OUTCOME_ERROR_HANDLED:
            return "OUTCOME_ERROR_HANDLED";
        case exports.AwsLambdaTags_Outcome.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
exports.AwsLambdaTags_ResponseMode = void 0;
(function (AwsLambdaTags_ResponseMode) {
    /** RESPONSE_MODE_UNSPECIFIED - Mode not confirmed */
    AwsLambdaTags_ResponseMode[AwsLambdaTags_ResponseMode["RESPONSE_MODE_UNSPECIFIED"] = 0] = "RESPONSE_MODE_UNSPECIFIED";
    /** RESPONSE_MODE_BUFFERED - Regular buffered response */
    AwsLambdaTags_ResponseMode[AwsLambdaTags_ResponseMode["RESPONSE_MODE_BUFFERED"] = 1] = "RESPONSE_MODE_BUFFERED";
    /** RESPONSE_MODE_RESPONSE_STREAM - Stream response */
    AwsLambdaTags_ResponseMode[AwsLambdaTags_ResponseMode["RESPONSE_MODE_RESPONSE_STREAM"] = 2] = "RESPONSE_MODE_RESPONSE_STREAM";
    AwsLambdaTags_ResponseMode[AwsLambdaTags_ResponseMode["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(exports.AwsLambdaTags_ResponseMode || (exports.AwsLambdaTags_ResponseMode = {}));
function awsLambdaTags_ResponseModeFromJSON(object) {
    switch (object) {
        case 0:
        case "RESPONSE_MODE_UNSPECIFIED":
            return exports.AwsLambdaTags_ResponseMode.RESPONSE_MODE_UNSPECIFIED;
        case 1:
        case "RESPONSE_MODE_BUFFERED":
            return exports.AwsLambdaTags_ResponseMode.RESPONSE_MODE_BUFFERED;
        case 2:
        case "RESPONSE_MODE_RESPONSE_STREAM":
            return exports.AwsLambdaTags_ResponseMode.RESPONSE_MODE_RESPONSE_STREAM;
        case -1:
        case "UNRECOGNIZED":
        default:
            return exports.AwsLambdaTags_ResponseMode.UNRECOGNIZED;
    }
}
function awsLambdaTags_ResponseModeToJSON(object) {
    switch (object) {
        case exports.AwsLambdaTags_ResponseMode.RESPONSE_MODE_UNSPECIFIED:
            return "RESPONSE_MODE_UNSPECIFIED";
        case exports.AwsLambdaTags_ResponseMode.RESPONSE_MODE_BUFFERED:
            return "RESPONSE_MODE_BUFFERED";
        case exports.AwsLambdaTags_ResponseMode.RESPONSE_MODE_RESPONSE_STREAM:
            return "RESPONSE_MODE_RESPONSE_STREAM";
        case exports.AwsLambdaTags_ResponseMode.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
function createBaseAwsTags() {
    return {
        lambda: undefined,
        sdk: undefined,
        accountId: undefined,
        region: undefined,
        requestId: undefined,
        resourceName: undefined,
        sequenceId: undefined,
        logGroup: undefined,
        logStream: undefined,
    };
}
const AwsTags = {
    encode(message, writer = _m0__default["default"].Writer.create()) {
        if (message.lambda !== undefined) {
            AwsLambdaTags.encode(message.lambda, writer.uint32(802).fork()).ldelim();
        }
        if (message.sdk !== undefined) {
            AwsSdkTags.encode(message.sdk, writer.uint32(810).fork()).ldelim();
        }
        if (message.accountId !== undefined) {
            writer.uint32(818).string(message.accountId);
        }
        if (message.region !== undefined) {
            writer.uint32(826).string(message.region);
        }
        if (message.requestId !== undefined) {
            writer.uint32(834).string(message.requestId);
        }
        if (message.resourceName !== undefined) {
            writer.uint32(842).string(message.resourceName);
        }
        if (message.sequenceId !== undefined) {
            writer.uint32(26).string(message.sequenceId);
        }
        if (message.logGroup !== undefined) {
            writer.uint32(34).string(message.logGroup);
        }
        if (message.logStream !== undefined) {
            writer.uint32(42).string(message.logStream);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0__default["default"].Reader ? input : _m0__default["default"].Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAwsTags();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 100:
                    if (tag !== 802) {
                        break;
                    }
                    message.lambda = AwsLambdaTags.decode(reader, reader.uint32());
                    continue;
                case 101:
                    if (tag !== 810) {
                        break;
                    }
                    message.sdk = AwsSdkTags.decode(reader, reader.uint32());
                    continue;
                case 102:
                    if (tag !== 818) {
                        break;
                    }
                    message.accountId = reader.string();
                    continue;
                case 103:
                    if (tag !== 826) {
                        break;
                    }
                    message.region = reader.string();
                    continue;
                case 104:
                    if (tag !== 834) {
                        break;
                    }
                    message.requestId = reader.string();
                    continue;
                case 105:
                    if (tag !== 842) {
                        break;
                    }
                    message.resourceName = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.sequenceId = reader.string();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.logGroup = reader.string();
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.logStream = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            lambda: isSet(object.lambda) ? AwsLambdaTags.fromJSON(object.lambda) : undefined,
            sdk: isSet(object.sdk) ? AwsSdkTags.fromJSON(object.sdk) : undefined,
            accountId: isSet(object.accountId) ? String(object.accountId) : undefined,
            region: isSet(object.region) ? String(object.region) : undefined,
            requestId: isSet(object.requestId) ? String(object.requestId) : undefined,
            resourceName: isSet(object.resourceName) ? String(object.resourceName) : undefined,
            sequenceId: isSet(object.sequenceId) ? String(object.sequenceId) : undefined,
            logGroup: isSet(object.logGroup) ? String(object.logGroup) : undefined,
            logStream: isSet(object.logStream) ? String(object.logStream) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        message.lambda !== undefined && (obj.lambda = message.lambda ? AwsLambdaTags.toJSON(message.lambda) : undefined);
        message.sdk !== undefined && (obj.sdk = message.sdk ? AwsSdkTags.toJSON(message.sdk) : undefined);
        message.accountId !== undefined && (obj.accountId = message.accountId);
        message.region !== undefined && (obj.region = message.region);
        message.requestId !== undefined && (obj.requestId = message.requestId);
        message.resourceName !== undefined && (obj.resourceName = message.resourceName);
        message.sequenceId !== undefined && (obj.sequenceId = message.sequenceId);
        message.logGroup !== undefined && (obj.logGroup = message.logGroup);
        message.logStream !== undefined && (obj.logStream = message.logStream);
        return obj;
    },
    create(base) {
        return AwsTags.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c, _d, _e, _f, _g;
        const message = createBaseAwsTags();
        message.lambda = (object.lambda !== undefined && object.lambda !== null)
            ? AwsLambdaTags.fromPartial(object.lambda)
            : undefined;
        message.sdk = (object.sdk !== undefined && object.sdk !== null) ? AwsSdkTags.fromPartial(object.sdk) : undefined;
        message.accountId = (_a = object.accountId) !== null && _a !== void 0 ? _a : undefined;
        message.region = (_b = object.region) !== null && _b !== void 0 ? _b : undefined;
        message.requestId = (_c = object.requestId) !== null && _c !== void 0 ? _c : undefined;
        message.resourceName = (_d = object.resourceName) !== null && _d !== void 0 ? _d : undefined;
        message.sequenceId = (_e = object.sequenceId) !== null && _e !== void 0 ? _e : undefined;
        message.logGroup = (_f = object.logGroup) !== null && _f !== void 0 ? _f : undefined;
        message.logStream = (_g = object.logStream) !== null && _g !== void 0 ? _g : undefined;
        return message;
    },
};
function createBaseAwsApiGatewayTags() {
    return { accountId: "", apiId: "", apiStage: "", request: undefined };
}
const AwsApiGatewayTags = {
    encode(message, writer = _m0__default["default"].Writer.create()) {
        if (message.accountId !== "") {
            writer.uint32(10).string(message.accountId);
        }
        if (message.apiId !== "") {
            writer.uint32(18).string(message.apiId);
        }
        if (message.apiStage !== "") {
            writer.uint32(26).string(message.apiStage);
        }
        if (message.request !== undefined) {
            AwsApiGatewayTags_AwsApiGatewayRequestTags.encode(message.request, writer.uint32(122).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0__default["default"].Reader ? input : _m0__default["default"].Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAwsApiGatewayTags();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.accountId = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.apiId = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.apiStage = reader.string();
                    continue;
                case 15:
                    if (tag !== 122) {
                        break;
                    }
                    message.request = AwsApiGatewayTags_AwsApiGatewayRequestTags.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            accountId: isSet(object.accountId) ? String(object.accountId) : "",
            apiId: isSet(object.apiId) ? String(object.apiId) : "",
            apiStage: isSet(object.apiStage) ? String(object.apiStage) : "",
            request: isSet(object.request) ? AwsApiGatewayTags_AwsApiGatewayRequestTags.fromJSON(object.request) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        message.accountId !== undefined && (obj.accountId = message.accountId);
        message.apiId !== undefined && (obj.apiId = message.apiId);
        message.apiStage !== undefined && (obj.apiStage = message.apiStage);
        message.request !== undefined &&
            (obj.request = message.request ? AwsApiGatewayTags_AwsApiGatewayRequestTags.toJSON(message.request) : undefined);
        return obj;
    },
    create(base) {
        return AwsApiGatewayTags.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c;
        const message = createBaseAwsApiGatewayTags();
        message.accountId = (_a = object.accountId) !== null && _a !== void 0 ? _a : "";
        message.apiId = (_b = object.apiId) !== null && _b !== void 0 ? _b : "";
        message.apiStage = (_c = object.apiStage) !== null && _c !== void 0 ? _c : "";
        message.request = (object.request !== undefined && object.request !== null)
            ? AwsApiGatewayTags_AwsApiGatewayRequestTags.fromPartial(object.request)
            : undefined;
        return message;
    },
};
function createBaseAwsApiGatewayTags_AwsApiGatewayRequestTags() {
    return { id: "", timeEpoch: Long__default["default"].UZERO, pathParameterNames: [] };
}
const AwsApiGatewayTags_AwsApiGatewayRequestTags = {
    encode(message, writer = _m0__default["default"].Writer.create()) {
        if (message.id !== "") {
            writer.uint32(10).string(message.id);
        }
        if (!message.timeEpoch.isZero()) {
            writer.uint32(16).uint64(message.timeEpoch);
        }
        for (const v of message.pathParameterNames) {
            writer.uint32(34).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0__default["default"].Reader ? input : _m0__default["default"].Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAwsApiGatewayTags_AwsApiGatewayRequestTags();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.id = reader.string();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.timeEpoch = reader.uint64();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.pathParameterNames.push(reader.string());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            id: isSet(object.id) ? String(object.id) : "",
            timeEpoch: isSet(object.timeEpoch) ? Long__default["default"].fromValue(object.timeEpoch) : Long__default["default"].UZERO,
            pathParameterNames: Array.isArray(object === null || object === void 0 ? void 0 : object.pathParameterNames)
                ? object.pathParameterNames.map((e) => String(e))
                : [],
        };
    },
    toJSON(message) {
        const obj = {};
        message.id !== undefined && (obj.id = message.id);
        message.timeEpoch !== undefined && (obj.timeEpoch = (message.timeEpoch || Long__default["default"].UZERO).toString());
        if (message.pathParameterNames) {
            obj.pathParameterNames = message.pathParameterNames.map((e) => e);
        }
        else {
            obj.pathParameterNames = [];
        }
        return obj;
    },
    create(base) {
        return AwsApiGatewayTags_AwsApiGatewayRequestTags.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseAwsApiGatewayTags_AwsApiGatewayRequestTags();
        message.id = (_a = object.id) !== null && _a !== void 0 ? _a : "";
        message.timeEpoch = (object.timeEpoch !== undefined && object.timeEpoch !== null)
            ? Long__default["default"].fromValue(object.timeEpoch)
            : Long__default["default"].UZERO;
        message.pathParameterNames = ((_b = object.pathParameterNames) === null || _b === void 0 ? void 0 : _b.map((e) => e)) || [];
        return message;
    },
};
function createBaseAwsLambdaTags() {
    return {
        arch: "",
        isColdstart: false,
        eventType: undefined,
        eventSource: undefined,
        logGroup: undefined,
        logStreamName: undefined,
        maxMemory: undefined,
        name: "",
        requestId: "",
        version: "",
        accountId: undefined,
        outcome: 0,
        errorExceptionMessage: undefined,
        errorExceptionStacktrace: undefined,
        duration: undefined,
        requestBody: undefined,
        responseBody: undefined,
        logsStartTimeUnix: undefined,
        logsEndTimeUnix: undefined,
        responseMode: undefined,
        sqs: undefined,
        sns: undefined,
        http: undefined,
        apiGateway: undefined,
        httpRouter: undefined,
        initialization: undefined,
        invocation: undefined,
        runtime: undefined,
    };
}
const AwsLambdaTags = {
    encode(message, writer = _m0__default["default"].Writer.create()) {
        if (message.arch !== "") {
            writer.uint32(10).string(message.arch);
        }
        if (message.isColdstart === true) {
            writer.uint32(16).bool(message.isColdstart);
        }
        if (message.eventType !== undefined) {
            writer.uint32(26).string(message.eventType);
        }
        if (message.eventSource !== undefined) {
            writer.uint32(34).string(message.eventSource);
        }
        if (message.logGroup !== undefined) {
            writer.uint32(42).string(message.logGroup);
        }
        if (message.logStreamName !== undefined) {
            writer.uint32(50).string(message.logStreamName);
        }
        if (message.maxMemory !== undefined) {
            writer.uint32(56).uint32(message.maxMemory);
        }
        if (message.name !== "") {
            writer.uint32(66).string(message.name);
        }
        if (message.requestId !== "") {
            writer.uint32(74).string(message.requestId);
        }
        if (message.version !== "") {
            writer.uint32(90).string(message.version);
        }
        if (message.accountId !== undefined) {
            writer.uint32(98).string(message.accountId);
        }
        if (message.outcome !== 0) {
            writer.uint32(112).int32(message.outcome);
        }
        if (message.errorExceptionMessage !== undefined) {
            writer.uint32(122).string(message.errorExceptionMessage);
        }
        if (message.errorExceptionStacktrace !== undefined) {
            writer.uint32(130).string(message.errorExceptionStacktrace);
        }
        if (message.duration !== undefined) {
            writer.uint32(136).uint32(message.duration);
        }
        if (message.requestBody !== undefined) {
            writer.uint32(146).string(message.requestBody);
        }
        if (message.responseBody !== undefined) {
            writer.uint32(154).string(message.responseBody);
        }
        if (message.logsStartTimeUnix !== undefined) {
            writer.uint32(160).uint64(message.logsStartTimeUnix);
        }
        if (message.logsEndTimeUnix !== undefined) {
            writer.uint32(168).uint64(message.logsEndTimeUnix);
        }
        if (message.responseMode !== undefined) {
            writer.uint32(176).int32(message.responseMode);
        }
        if (message.sqs !== undefined) {
            AwsSqsEventTags.encode(message.sqs, writer.uint32(802).fork()).ldelim();
        }
        if (message.sns !== undefined) {
            AwsSnsEventTags.encode(message.sns, writer.uint32(810).fork()).ldelim();
        }
        if (message.http !== undefined) {
            HttpTags.encode(message.http, writer.uint32(818).fork()).ldelim();
        }
        if (message.apiGateway !== undefined) {
            AwsApiGatewayTags.encode(message.apiGateway, writer.uint32(826).fork()).ldelim();
        }
        if (message.httpRouter !== undefined) {
            HttpRouterTags.encode(message.httpRouter, writer.uint32(850).fork()).ldelim();
        }
        if (message.initialization !== undefined) {
            AwsLambdaInitializationTags.encode(message.initialization, writer.uint32(834).fork()).ldelim();
        }
        if (message.invocation !== undefined) {
            AwsLambdaInvocationTags.encode(message.invocation, writer.uint32(842).fork()).ldelim();
        }
        if (message.runtime !== undefined) {
            AwsLambdaRuntimeTags.encode(message.runtime, writer.uint32(858).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0__default["default"].Reader ? input : _m0__default["default"].Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAwsLambdaTags();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.arch = reader.string();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.isColdstart = reader.bool();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.eventType = reader.string();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.eventSource = reader.string();
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.logGroup = reader.string();
                    continue;
                case 6:
                    if (tag !== 50) {
                        break;
                    }
                    message.logStreamName = reader.string();
                    continue;
                case 7:
                    if (tag !== 56) {
                        break;
                    }
                    message.maxMemory = reader.uint32();
                    continue;
                case 8:
                    if (tag !== 66) {
                        break;
                    }
                    message.name = reader.string();
                    continue;
                case 9:
                    if (tag !== 74) {
                        break;
                    }
                    message.requestId = reader.string();
                    continue;
                case 11:
                    if (tag !== 90) {
                        break;
                    }
                    message.version = reader.string();
                    continue;
                case 12:
                    if (tag !== 98) {
                        break;
                    }
                    message.accountId = reader.string();
                    continue;
                case 14:
                    if (tag !== 112) {
                        break;
                    }
                    message.outcome = reader.int32();
                    continue;
                case 15:
                    if (tag !== 122) {
                        break;
                    }
                    message.errorExceptionMessage = reader.string();
                    continue;
                case 16:
                    if (tag !== 130) {
                        break;
                    }
                    message.errorExceptionStacktrace = reader.string();
                    continue;
                case 17:
                    if (tag !== 136) {
                        break;
                    }
                    message.duration = reader.uint32();
                    continue;
                case 18:
                    if (tag !== 146) {
                        break;
                    }
                    message.requestBody = reader.string();
                    continue;
                case 19:
                    if (tag !== 154) {
                        break;
                    }
                    message.responseBody = reader.string();
                    continue;
                case 20:
                    if (tag !== 160) {
                        break;
                    }
                    message.logsStartTimeUnix = reader.uint64();
                    continue;
                case 21:
                    if (tag !== 168) {
                        break;
                    }
                    message.logsEndTimeUnix = reader.uint64();
                    continue;
                case 22:
                    if (tag !== 176) {
                        break;
                    }
                    message.responseMode = reader.int32();
                    continue;
                case 100:
                    if (tag !== 802) {
                        break;
                    }
                    message.sqs = AwsSqsEventTags.decode(reader, reader.uint32());
                    continue;
                case 101:
                    if (tag !== 810) {
                        break;
                    }
                    message.sns = AwsSnsEventTags.decode(reader, reader.uint32());
                    continue;
                case 102:
                    if (tag !== 818) {
                        break;
                    }
                    message.http = HttpTags.decode(reader, reader.uint32());
                    continue;
                case 103:
                    if (tag !== 826) {
                        break;
                    }
                    message.apiGateway = AwsApiGatewayTags.decode(reader, reader.uint32());
                    continue;
                case 106:
                    if (tag !== 850) {
                        break;
                    }
                    message.httpRouter = HttpRouterTags.decode(reader, reader.uint32());
                    continue;
                case 104:
                    if (tag !== 834) {
                        break;
                    }
                    message.initialization = AwsLambdaInitializationTags.decode(reader, reader.uint32());
                    continue;
                case 105:
                    if (tag !== 842) {
                        break;
                    }
                    message.invocation = AwsLambdaInvocationTags.decode(reader, reader.uint32());
                    continue;
                case 107:
                    if (tag !== 858) {
                        break;
                    }
                    message.runtime = AwsLambdaRuntimeTags.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            arch: isSet(object.arch) ? String(object.arch) : "",
            isColdstart: isSet(object.isColdstart) ? Boolean(object.isColdstart) : false,
            eventType: isSet(object.eventType) ? String(object.eventType) : undefined,
            eventSource: isSet(object.eventSource) ? String(object.eventSource) : undefined,
            logGroup: isSet(object.logGroup) ? String(object.logGroup) : undefined,
            logStreamName: isSet(object.logStreamName) ? String(object.logStreamName) : undefined,
            maxMemory: isSet(object.maxMemory) ? Number(object.maxMemory) : undefined,
            name: isSet(object.name) ? String(object.name) : "",
            requestId: isSet(object.requestId) ? String(object.requestId) : "",
            version: isSet(object.version) ? String(object.version) : "",
            accountId: isSet(object.accountId) ? String(object.accountId) : undefined,
            outcome: isSet(object.outcome) ? awsLambdaTags_OutcomeFromJSON(object.outcome) : 0,
            errorExceptionMessage: isSet(object.errorExceptionMessage) ? String(object.errorExceptionMessage) : undefined,
            errorExceptionStacktrace: isSet(object.errorExceptionStacktrace)
                ? String(object.errorExceptionStacktrace)
                : undefined,
            duration: isSet(object.duration) ? Number(object.duration) : undefined,
            requestBody: isSet(object.requestBody) ? String(object.requestBody) : undefined,
            responseBody: isSet(object.responseBody) ? String(object.responseBody) : undefined,
            logsStartTimeUnix: isSet(object.logsStartTimeUnix) ? Long__default["default"].fromValue(object.logsStartTimeUnix) : undefined,
            logsEndTimeUnix: isSet(object.logsEndTimeUnix) ? Long__default["default"].fromValue(object.logsEndTimeUnix) : undefined,
            responseMode: isSet(object.responseMode) ? awsLambdaTags_ResponseModeFromJSON(object.responseMode) : undefined,
            sqs: isSet(object.sqs) ? AwsSqsEventTags.fromJSON(object.sqs) : undefined,
            sns: isSet(object.sns) ? AwsSnsEventTags.fromJSON(object.sns) : undefined,
            http: isSet(object.http) ? HttpTags.fromJSON(object.http) : undefined,
            apiGateway: isSet(object.apiGateway) ? AwsApiGatewayTags.fromJSON(object.apiGateway) : undefined,
            httpRouter: isSet(object.httpRouter) ? HttpRouterTags.fromJSON(object.httpRouter) : undefined,
            initialization: isSet(object.initialization)
                ? AwsLambdaInitializationTags.fromJSON(object.initialization)
                : undefined,
            invocation: isSet(object.invocation) ? AwsLambdaInvocationTags.fromJSON(object.invocation) : undefined,
            runtime: isSet(object.runtime) ? AwsLambdaRuntimeTags.fromJSON(object.runtime) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        message.arch !== undefined && (obj.arch = message.arch);
        message.isColdstart !== undefined && (obj.isColdstart = message.isColdstart);
        message.eventType !== undefined && (obj.eventType = message.eventType);
        message.eventSource !== undefined && (obj.eventSource = message.eventSource);
        message.logGroup !== undefined && (obj.logGroup = message.logGroup);
        message.logStreamName !== undefined && (obj.logStreamName = message.logStreamName);
        message.maxMemory !== undefined && (obj.maxMemory = Math.round(message.maxMemory));
        message.name !== undefined && (obj.name = message.name);
        message.requestId !== undefined && (obj.requestId = message.requestId);
        message.version !== undefined && (obj.version = message.version);
        message.accountId !== undefined && (obj.accountId = message.accountId);
        message.outcome !== undefined && (obj.outcome = awsLambdaTags_OutcomeToJSON(message.outcome));
        message.errorExceptionMessage !== undefined && (obj.errorExceptionMessage = message.errorExceptionMessage);
        message.errorExceptionStacktrace !== undefined && (obj.errorExceptionStacktrace = message.errorExceptionStacktrace);
        message.duration !== undefined && (obj.duration = Math.round(message.duration));
        message.requestBody !== undefined && (obj.requestBody = message.requestBody);
        message.responseBody !== undefined && (obj.responseBody = message.responseBody);
        message.logsStartTimeUnix !== undefined &&
            (obj.logsStartTimeUnix = (message.logsStartTimeUnix || undefined).toString());
        message.logsEndTimeUnix !== undefined && (obj.logsEndTimeUnix = (message.logsEndTimeUnix || undefined).toString());
        message.responseMode !== undefined && (obj.responseMode = message.responseMode !== undefined
            ? awsLambdaTags_ResponseModeToJSON(message.responseMode)
            : undefined);
        message.sqs !== undefined && (obj.sqs = message.sqs ? AwsSqsEventTags.toJSON(message.sqs) : undefined);
        message.sns !== undefined && (obj.sns = message.sns ? AwsSnsEventTags.toJSON(message.sns) : undefined);
        message.http !== undefined && (obj.http = message.http ? HttpTags.toJSON(message.http) : undefined);
        message.apiGateway !== undefined &&
            (obj.apiGateway = message.apiGateway ? AwsApiGatewayTags.toJSON(message.apiGateway) : undefined);
        message.httpRouter !== undefined &&
            (obj.httpRouter = message.httpRouter ? HttpRouterTags.toJSON(message.httpRouter) : undefined);
        message.initialization !== undefined && (obj.initialization = message.initialization
            ? AwsLambdaInitializationTags.toJSON(message.initialization)
            : undefined);
        message.invocation !== undefined &&
            (obj.invocation = message.invocation ? AwsLambdaInvocationTags.toJSON(message.invocation) : undefined);
        message.runtime !== undefined &&
            (obj.runtime = message.runtime ? AwsLambdaRuntimeTags.toJSON(message.runtime) : undefined);
        return obj;
    },
    create(base) {
        return AwsLambdaTags.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k, _l, _m, _o, _p, _q, _r, _s, _t;
        const message = createBaseAwsLambdaTags();
        message.arch = (_a = object.arch) !== null && _a !== void 0 ? _a : "";
        message.isColdstart = (_b = object.isColdstart) !== null && _b !== void 0 ? _b : false;
        message.eventType = (_c = object.eventType) !== null && _c !== void 0 ? _c : undefined;
        message.eventSource = (_d = object.eventSource) !== null && _d !== void 0 ? _d : undefined;
        message.logGroup = (_e = object.logGroup) !== null && _e !== void 0 ? _e : undefined;
        message.logStreamName = (_f = object.logStreamName) !== null && _f !== void 0 ? _f : undefined;
        message.maxMemory = (_g = object.maxMemory) !== null && _g !== void 0 ? _g : undefined;
        message.name = (_h = object.name) !== null && _h !== void 0 ? _h : "";
        message.requestId = (_j = object.requestId) !== null && _j !== void 0 ? _j : "";
        message.version = (_k = object.version) !== null && _k !== void 0 ? _k : "";
        message.accountId = (_l = object.accountId) !== null && _l !== void 0 ? _l : undefined;
        message.outcome = (_m = object.outcome) !== null && _m !== void 0 ? _m : 0;
        message.errorExceptionMessage = (_o = object.errorExceptionMessage) !== null && _o !== void 0 ? _o : undefined;
        message.errorExceptionStacktrace = (_p = object.errorExceptionStacktrace) !== null && _p !== void 0 ? _p : undefined;
        message.duration = (_q = object.duration) !== null && _q !== void 0 ? _q : undefined;
        message.requestBody = (_r = object.requestBody) !== null && _r !== void 0 ? _r : undefined;
        message.responseBody = (_s = object.responseBody) !== null && _s !== void 0 ? _s : undefined;
        message.logsStartTimeUnix = (object.logsStartTimeUnix !== undefined && object.logsStartTimeUnix !== null)
            ? Long__default["default"].fromValue(object.logsStartTimeUnix)
            : undefined;
        message.logsEndTimeUnix = (object.logsEndTimeUnix !== undefined && object.logsEndTimeUnix !== null)
            ? Long__default["default"].fromValue(object.logsEndTimeUnix)
            : undefined;
        message.responseMode = (_t = object.responseMode) !== null && _t !== void 0 ? _t : undefined;
        message.sqs = (object.sqs !== undefined && object.sqs !== null)
            ? AwsSqsEventTags.fromPartial(object.sqs)
            : undefined;
        message.sns = (object.sns !== undefined && object.sns !== null)
            ? AwsSnsEventTags.fromPartial(object.sns)
            : undefined;
        message.http = (object.http !== undefined && object.http !== null) ? HttpTags.fromPartial(object.http) : undefined;
        message.apiGateway = (object.apiGateway !== undefined && object.apiGateway !== null)
            ? AwsApiGatewayTags.fromPartial(object.apiGateway)
            : undefined;
        message.httpRouter = (object.httpRouter !== undefined && object.httpRouter !== null)
            ? HttpRouterTags.fromPartial(object.httpRouter)
            : undefined;
        message.initialization = (object.initialization !== undefined && object.initialization !== null)
            ? AwsLambdaInitializationTags.fromPartial(object.initialization)
            : undefined;
        message.invocation = (object.invocation !== undefined && object.invocation !== null)
            ? AwsLambdaInvocationTags.fromPartial(object.invocation)
            : undefined;
        message.runtime = (object.runtime !== undefined && object.runtime !== null)
            ? AwsLambdaRuntimeTags.fromPartial(object.runtime)
            : undefined;
        return message;
    },
};
function createBaseHttpRouterTags() {
    return { path: "" };
}
const HttpRouterTags = {
    encode(message, writer = _m0__default["default"].Writer.create()) {
        if (message.path !== "") {
            writer.uint32(10).string(message.path);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0__default["default"].Reader ? input : _m0__default["default"].Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseHttpRouterTags();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.path = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { path: isSet(object.path) ? String(object.path) : "" };
    },
    toJSON(message) {
        const obj = {};
        message.path !== undefined && (obj.path = message.path);
        return obj;
    },
    create(base) {
        return HttpRouterTags.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseHttpRouterTags();
        message.path = (_a = object.path) !== null && _a !== void 0 ? _a : "";
        return message;
    },
};
function createBaseAwsSqsEventTags() {
    return { queueName: "", messageIds: [] };
}
const AwsSqsEventTags = {
    encode(message, writer = _m0__default["default"].Writer.create()) {
        if (message.queueName !== "") {
            writer.uint32(10).string(message.queueName);
        }
        for (const v of message.messageIds) {
            writer.uint32(18).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0__default["default"].Reader ? input : _m0__default["default"].Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAwsSqsEventTags();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.queueName = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.messageIds.push(reader.string());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            queueName: isSet(object.queueName) ? String(object.queueName) : "",
            messageIds: Array.isArray(object === null || object === void 0 ? void 0 : object.messageIds) ? object.messageIds.map((e) => String(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        message.queueName !== undefined && (obj.queueName = message.queueName);
        if (message.messageIds) {
            obj.messageIds = message.messageIds.map((e) => e);
        }
        else {
            obj.messageIds = [];
        }
        return obj;
    },
    create(base) {
        return AwsSqsEventTags.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseAwsSqsEventTags();
        message.queueName = (_a = object.queueName) !== null && _a !== void 0 ? _a : "";
        message.messageIds = ((_b = object.messageIds) === null || _b === void 0 ? void 0 : _b.map((e) => e)) || [];
        return message;
    },
};
function createBaseAwsSnsEventTags() {
    return { topicName: "", messageIds: [] };
}
const AwsSnsEventTags = {
    encode(message, writer = _m0__default["default"].Writer.create()) {
        if (message.topicName !== "") {
            writer.uint32(10).string(message.topicName);
        }
        for (const v of message.messageIds) {
            writer.uint32(18).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0__default["default"].Reader ? input : _m0__default["default"].Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAwsSnsEventTags();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.topicName = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.messageIds.push(reader.string());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            topicName: isSet(object.topicName) ? String(object.topicName) : "",
            messageIds: Array.isArray(object === null || object === void 0 ? void 0 : object.messageIds) ? object.messageIds.map((e) => String(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        message.topicName !== undefined && (obj.topicName = message.topicName);
        if (message.messageIds) {
            obj.messageIds = message.messageIds.map((e) => e);
        }
        else {
            obj.messageIds = [];
        }
        return obj;
    },
    create(base) {
        return AwsSnsEventTags.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseAwsSnsEventTags();
        message.topicName = (_a = object.topicName) !== null && _a !== void 0 ? _a : "";
        message.messageIds = ((_b = object.messageIds) === null || _b === void 0 ? void 0 : _b.map((e) => e)) || [];
        return message;
    },
};
function createBaseAwsLambdaRuntimeTags() {
    return { identifier: "", version: "", arn: "" };
}
const AwsLambdaRuntimeTags = {
    encode(message, writer = _m0__default["default"].Writer.create()) {
        if (message.identifier !== "") {
            writer.uint32(10).string(message.identifier);
        }
        if (message.version !== "") {
            writer.uint32(18).string(message.version);
        }
        if (message.arn !== "") {
            writer.uint32(26).string(message.arn);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0__default["default"].Reader ? input : _m0__default["default"].Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAwsLambdaRuntimeTags();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.identifier = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.version = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.arn = reader.string();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            identifier: isSet(object.identifier) ? String(object.identifier) : "",
            version: isSet(object.version) ? String(object.version) : "",
            arn: isSet(object.arn) ? String(object.arn) : "",
        };
    },
    toJSON(message) {
        const obj = {};
        message.identifier !== undefined && (obj.identifier = message.identifier);
        message.version !== undefined && (obj.version = message.version);
        message.arn !== undefined && (obj.arn = message.arn);
        return obj;
    },
    create(base) {
        return AwsLambdaRuntimeTags.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c;
        const message = createBaseAwsLambdaRuntimeTags();
        message.identifier = (_a = object.identifier) !== null && _a !== void 0 ? _a : "";
        message.version = (_b = object.version) !== null && _b !== void 0 ? _b : "";
        message.arn = (_c = object.arn) !== null && _c !== void 0 ? _c : "";
        return message;
    },
};
function createBaseAwsLambdaInitializationTags() {
    return { initializationDuration: 0 };
}
const AwsLambdaInitializationTags = {
    encode(message, writer = _m0__default["default"].Writer.create()) {
        if (message.initializationDuration !== 0) {
            writer.uint32(8).uint32(message.initializationDuration);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0__default["default"].Reader ? input : _m0__default["default"].Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAwsLambdaInitializationTags();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.initializationDuration = reader.uint32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { initializationDuration: isSet(object.initializationDuration) ? Number(object.initializationDuration) : 0 };
    },
    toJSON(message) {
        const obj = {};
        message.initializationDuration !== undefined &&
            (obj.initializationDuration = Math.round(message.initializationDuration));
        return obj;
    },
    create(base) {
        return AwsLambdaInitializationTags.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseAwsLambdaInitializationTags();
        message.initializationDuration = (_a = object.initializationDuration) !== null && _a !== void 0 ? _a : 0;
        return message;
    },
};
function createBaseAwsLambdaInvocationTags() {
    return { invocationDuration: 0 };
}
const AwsLambdaInvocationTags = {
    encode(message, writer = _m0__default["default"].Writer.create()) {
        if (message.invocationDuration !== 0) {
            writer.uint32(8).uint32(message.invocationDuration);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0__default["default"].Reader ? input : _m0__default["default"].Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAwsLambdaInvocationTags();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 8) {
                        break;
                    }
                    message.invocationDuration = reader.uint32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return { invocationDuration: isSet(object.invocationDuration) ? Number(object.invocationDuration) : 0 };
    },
    toJSON(message) {
        const obj = {};
        message.invocationDuration !== undefined && (obj.invocationDuration = Math.round(message.invocationDuration));
        return obj;
    },
    create(base) {
        return AwsLambdaInvocationTags.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a;
        const message = createBaseAwsLambdaInvocationTags();
        message.invocationDuration = (_a = object.invocationDuration) !== null && _a !== void 0 ? _a : 0;
        return message;
    },
};
function createBaseAwsSdkTags() {
    return {
        region: undefined,
        signatureVersion: undefined,
        service: "",
        operation: "",
        requestId: undefined,
        error: undefined,
        dynamodb: undefined,
        sqs: undefined,
        sns: undefined,
    };
}
const AwsSdkTags = {
    encode(message, writer = _m0__default["default"].Writer.create()) {
        if (message.region !== undefined) {
            writer.uint32(18).string(message.region);
        }
        if (message.signatureVersion !== undefined) {
            writer.uint32(26).string(message.signatureVersion);
        }
        if (message.service !== "") {
            writer.uint32(34).string(message.service);
        }
        if (message.operation !== "") {
            writer.uint32(42).string(message.operation);
        }
        if (message.requestId !== undefined) {
            writer.uint32(50).string(message.requestId);
        }
        if (message.error !== undefined) {
            writer.uint32(58).string(message.error);
        }
        if (message.dynamodb !== undefined) {
            AwsSdkDynamodbTags.encode(message.dynamodb, writer.uint32(802).fork()).ldelim();
        }
        if (message.sqs !== undefined) {
            AwsSdkSqsTags.encode(message.sqs, writer.uint32(810).fork()).ldelim();
        }
        if (message.sns !== undefined) {
            AwsSdkSnsTags.encode(message.sns, writer.uint32(818).fork()).ldelim();
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0__default["default"].Reader ? input : _m0__default["default"].Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAwsSdkTags();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.region = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.signatureVersion = reader.string();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.service = reader.string();
                    continue;
                case 5:
                    if (tag !== 42) {
                        break;
                    }
                    message.operation = reader.string();
                    continue;
                case 6:
                    if (tag !== 50) {
                        break;
                    }
                    message.requestId = reader.string();
                    continue;
                case 7:
                    if (tag !== 58) {
                        break;
                    }
                    message.error = reader.string();
                    continue;
                case 100:
                    if (tag !== 802) {
                        break;
                    }
                    message.dynamodb = AwsSdkDynamodbTags.decode(reader, reader.uint32());
                    continue;
                case 101:
                    if (tag !== 810) {
                        break;
                    }
                    message.sqs = AwsSdkSqsTags.decode(reader, reader.uint32());
                    continue;
                case 102:
                    if (tag !== 818) {
                        break;
                    }
                    message.sns = AwsSdkSnsTags.decode(reader, reader.uint32());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            region: isSet(object.region) ? String(object.region) : undefined,
            signatureVersion: isSet(object.signatureVersion) ? String(object.signatureVersion) : undefined,
            service: isSet(object.service) ? String(object.service) : "",
            operation: isSet(object.operation) ? String(object.operation) : "",
            requestId: isSet(object.requestId) ? String(object.requestId) : undefined,
            error: isSet(object.error) ? String(object.error) : undefined,
            dynamodb: isSet(object.dynamodb) ? AwsSdkDynamodbTags.fromJSON(object.dynamodb) : undefined,
            sqs: isSet(object.sqs) ? AwsSdkSqsTags.fromJSON(object.sqs) : undefined,
            sns: isSet(object.sns) ? AwsSdkSnsTags.fromJSON(object.sns) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        message.region !== undefined && (obj.region = message.region);
        message.signatureVersion !== undefined && (obj.signatureVersion = message.signatureVersion);
        message.service !== undefined && (obj.service = message.service);
        message.operation !== undefined && (obj.operation = message.operation);
        message.requestId !== undefined && (obj.requestId = message.requestId);
        message.error !== undefined && (obj.error = message.error);
        message.dynamodb !== undefined &&
            (obj.dynamodb = message.dynamodb ? AwsSdkDynamodbTags.toJSON(message.dynamodb) : undefined);
        message.sqs !== undefined && (obj.sqs = message.sqs ? AwsSdkSqsTags.toJSON(message.sqs) : undefined);
        message.sns !== undefined && (obj.sns = message.sns ? AwsSdkSnsTags.toJSON(message.sns) : undefined);
        return obj;
    },
    create(base) {
        return AwsSdkTags.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c, _d, _e, _f;
        const message = createBaseAwsSdkTags();
        message.region = (_a = object.region) !== null && _a !== void 0 ? _a : undefined;
        message.signatureVersion = (_b = object.signatureVersion) !== null && _b !== void 0 ? _b : undefined;
        message.service = (_c = object.service) !== null && _c !== void 0 ? _c : "";
        message.operation = (_d = object.operation) !== null && _d !== void 0 ? _d : "";
        message.requestId = (_e = object.requestId) !== null && _e !== void 0 ? _e : undefined;
        message.error = (_f = object.error) !== null && _f !== void 0 ? _f : undefined;
        message.dynamodb = (object.dynamodb !== undefined && object.dynamodb !== null)
            ? AwsSdkDynamodbTags.fromPartial(object.dynamodb)
            : undefined;
        message.sqs = (object.sqs !== undefined && object.sqs !== null) ? AwsSdkSqsTags.fromPartial(object.sqs) : undefined;
        message.sns = (object.sns !== undefined && object.sns !== null) ? AwsSdkSnsTags.fromPartial(object.sns) : undefined;
        return message;
    },
};
function createBaseAwsSdkDynamodbTags() {
    return {
        tableName: undefined,
        projection: undefined,
        scanForward: undefined,
        attributesToGet: [],
        consistentRead: undefined,
        indexName: undefined,
        limit: undefined,
        select: undefined,
        segment: undefined,
        totalSegments: undefined,
        filter: undefined,
        keyCondition: undefined,
        exclusiveStartKey: undefined,
        attributeValues: undefined,
        count: undefined,
        scannedCount: undefined,
    };
}
const AwsSdkDynamodbTags = {
    encode(message, writer = _m0__default["default"].Writer.create()) {
        if (message.tableName !== undefined) {
            writer.uint32(10).string(message.tableName);
        }
        if (message.projection !== undefined) {
            writer.uint32(18).string(message.projection);
        }
        if (message.scanForward !== undefined) {
            writer.uint32(24).bool(message.scanForward);
        }
        for (const v of message.attributesToGet) {
            writer.uint32(34).string(v);
        }
        if (message.consistentRead !== undefined) {
            writer.uint32(40).bool(message.consistentRead);
        }
        if (message.indexName !== undefined) {
            writer.uint32(50).string(message.indexName);
        }
        if (message.limit !== undefined) {
            writer.uint32(56).uint32(message.limit);
        }
        if (message.select !== undefined) {
            writer.uint32(66).string(message.select);
        }
        if (message.segment !== undefined) {
            writer.uint32(72).uint32(message.segment);
        }
        if (message.totalSegments !== undefined) {
            writer.uint32(80).uint64(message.totalSegments);
        }
        if (message.filter !== undefined) {
            writer.uint32(90).string(message.filter);
        }
        if (message.keyCondition !== undefined) {
            writer.uint32(98).string(message.keyCondition);
        }
        if (message.exclusiveStartKey !== undefined) {
            writer.uint32(106).string(message.exclusiveStartKey);
        }
        if (message.attributeValues !== undefined) {
            writer.uint32(114).string(message.attributeValues);
        }
        if (message.count !== undefined) {
            writer.uint32(800).uint64(message.count);
        }
        if (message.scannedCount !== undefined) {
            writer.uint32(808).uint64(message.scannedCount);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0__default["default"].Reader ? input : _m0__default["default"].Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAwsSdkDynamodbTags();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.tableName = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.projection = reader.string();
                    continue;
                case 3:
                    if (tag !== 24) {
                        break;
                    }
                    message.scanForward = reader.bool();
                    continue;
                case 4:
                    if (tag !== 34) {
                        break;
                    }
                    message.attributesToGet.push(reader.string());
                    continue;
                case 5:
                    if (tag !== 40) {
                        break;
                    }
                    message.consistentRead = reader.bool();
                    continue;
                case 6:
                    if (tag !== 50) {
                        break;
                    }
                    message.indexName = reader.string();
                    continue;
                case 7:
                    if (tag !== 56) {
                        break;
                    }
                    message.limit = reader.uint32();
                    continue;
                case 8:
                    if (tag !== 66) {
                        break;
                    }
                    message.select = reader.string();
                    continue;
                case 9:
                    if (tag !== 72) {
                        break;
                    }
                    message.segment = reader.uint32();
                    continue;
                case 10:
                    if (tag !== 80) {
                        break;
                    }
                    message.totalSegments = reader.uint64();
                    continue;
                case 11:
                    if (tag !== 90) {
                        break;
                    }
                    message.filter = reader.string();
                    continue;
                case 12:
                    if (tag !== 98) {
                        break;
                    }
                    message.keyCondition = reader.string();
                    continue;
                case 13:
                    if (tag !== 106) {
                        break;
                    }
                    message.exclusiveStartKey = reader.string();
                    continue;
                case 14:
                    if (tag !== 114) {
                        break;
                    }
                    message.attributeValues = reader.string();
                    continue;
                case 100:
                    if (tag !== 800) {
                        break;
                    }
                    message.count = reader.uint64();
                    continue;
                case 101:
                    if (tag !== 808) {
                        break;
                    }
                    message.scannedCount = reader.uint64();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            tableName: isSet(object.tableName) ? String(object.tableName) : undefined,
            projection: isSet(object.projection) ? String(object.projection) : undefined,
            scanForward: isSet(object.scanForward) ? Boolean(object.scanForward) : undefined,
            attributesToGet: Array.isArray(object === null || object === void 0 ? void 0 : object.attributesToGet) ? object.attributesToGet.map((e) => String(e)) : [],
            consistentRead: isSet(object.consistentRead) ? Boolean(object.consistentRead) : undefined,
            indexName: isSet(object.indexName) ? String(object.indexName) : undefined,
            limit: isSet(object.limit) ? Number(object.limit) : undefined,
            select: isSet(object.select) ? String(object.select) : undefined,
            segment: isSet(object.segment) ? Number(object.segment) : undefined,
            totalSegments: isSet(object.totalSegments) ? Long__default["default"].fromValue(object.totalSegments) : undefined,
            filter: isSet(object.filter) ? String(object.filter) : undefined,
            keyCondition: isSet(object.keyCondition) ? String(object.keyCondition) : undefined,
            exclusiveStartKey: isSet(object.exclusiveStartKey) ? String(object.exclusiveStartKey) : undefined,
            attributeValues: isSet(object.attributeValues) ? String(object.attributeValues) : undefined,
            count: isSet(object.count) ? Long__default["default"].fromValue(object.count) : undefined,
            scannedCount: isSet(object.scannedCount) ? Long__default["default"].fromValue(object.scannedCount) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        message.tableName !== undefined && (obj.tableName = message.tableName);
        message.projection !== undefined && (obj.projection = message.projection);
        message.scanForward !== undefined && (obj.scanForward = message.scanForward);
        if (message.attributesToGet) {
            obj.attributesToGet = message.attributesToGet.map((e) => e);
        }
        else {
            obj.attributesToGet = [];
        }
        message.consistentRead !== undefined && (obj.consistentRead = message.consistentRead);
        message.indexName !== undefined && (obj.indexName = message.indexName);
        message.limit !== undefined && (obj.limit = Math.round(message.limit));
        message.select !== undefined && (obj.select = message.select);
        message.segment !== undefined && (obj.segment = Math.round(message.segment));
        message.totalSegments !== undefined && (obj.totalSegments = (message.totalSegments || undefined).toString());
        message.filter !== undefined && (obj.filter = message.filter);
        message.keyCondition !== undefined && (obj.keyCondition = message.keyCondition);
        message.exclusiveStartKey !== undefined && (obj.exclusiveStartKey = message.exclusiveStartKey);
        message.attributeValues !== undefined && (obj.attributeValues = message.attributeValues);
        message.count !== undefined && (obj.count = (message.count || undefined).toString());
        message.scannedCount !== undefined && (obj.scannedCount = (message.scannedCount || undefined).toString());
        return obj;
    },
    create(base) {
        return AwsSdkDynamodbTags.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c, _d, _e, _f, _g, _h, _j, _k, _l, _m, _o;
        const message = createBaseAwsSdkDynamodbTags();
        message.tableName = (_a = object.tableName) !== null && _a !== void 0 ? _a : undefined;
        message.projection = (_b = object.projection) !== null && _b !== void 0 ? _b : undefined;
        message.scanForward = (_c = object.scanForward) !== null && _c !== void 0 ? _c : undefined;
        message.attributesToGet = ((_d = object.attributesToGet) === null || _d === void 0 ? void 0 : _d.map((e) => e)) || [];
        message.consistentRead = (_e = object.consistentRead) !== null && _e !== void 0 ? _e : undefined;
        message.indexName = (_f = object.indexName) !== null && _f !== void 0 ? _f : undefined;
        message.limit = (_g = object.limit) !== null && _g !== void 0 ? _g : undefined;
        message.select = (_h = object.select) !== null && _h !== void 0 ? _h : undefined;
        message.segment = (_j = object.segment) !== null && _j !== void 0 ? _j : undefined;
        message.totalSegments = (object.totalSegments !== undefined && object.totalSegments !== null)
            ? Long__default["default"].fromValue(object.totalSegments)
            : undefined;
        message.filter = (_k = object.filter) !== null && _k !== void 0 ? _k : undefined;
        message.keyCondition = (_l = object.keyCondition) !== null && _l !== void 0 ? _l : undefined;
        message.exclusiveStartKey = (_m = object.exclusiveStartKey) !== null && _m !== void 0 ? _m : undefined;
        message.attributeValues = (_o = object.attributeValues) !== null && _o !== void 0 ? _o : undefined;
        message.count = (object.count !== undefined && object.count !== null) ? Long__default["default"].fromValue(object.count) : undefined;
        message.scannedCount = (object.scannedCount !== undefined && object.scannedCount !== null)
            ? Long__default["default"].fromValue(object.scannedCount)
            : undefined;
        return message;
    },
};
function createBaseAwsSdkSqsTags() {
    return { queueName: undefined, messageIds: [] };
}
const AwsSdkSqsTags = {
    encode(message, writer = _m0__default["default"].Writer.create()) {
        if (message.queueName !== undefined) {
            writer.uint32(10).string(message.queueName);
        }
        for (const v of message.messageIds) {
            writer.uint32(18).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0__default["default"].Reader ? input : _m0__default["default"].Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAwsSdkSqsTags();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.queueName = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.messageIds.push(reader.string());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            queueName: isSet(object.queueName) ? String(object.queueName) : undefined,
            messageIds: Array.isArray(object === null || object === void 0 ? void 0 : object.messageIds) ? object.messageIds.map((e) => String(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        message.queueName !== undefined && (obj.queueName = message.queueName);
        if (message.messageIds) {
            obj.messageIds = message.messageIds.map((e) => e);
        }
        else {
            obj.messageIds = [];
        }
        return obj;
    },
    create(base) {
        return AwsSdkSqsTags.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseAwsSdkSqsTags();
        message.queueName = (_a = object.queueName) !== null && _a !== void 0 ? _a : undefined;
        message.messageIds = ((_b = object.messageIds) === null || _b === void 0 ? void 0 : _b.map((e) => e)) || [];
        return message;
    },
};
function createBaseAwsSdkSnsTags() {
    return { topicName: undefined, messageIds: [] };
}
const AwsSdkSnsTags = {
    encode(message, writer = _m0__default["default"].Writer.create()) {
        if (message.topicName !== undefined) {
            writer.uint32(10).string(message.topicName);
        }
        for (const v of message.messageIds) {
            writer.uint32(18).string(v);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0__default["default"].Reader ? input : _m0__default["default"].Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseAwsSdkSnsTags();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.topicName = reader.string();
                    continue;
                case 2:
                    if (tag !== 18) {
                        break;
                    }
                    message.messageIds.push(reader.string());
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            topicName: isSet(object.topicName) ? String(object.topicName) : undefined,
            messageIds: Array.isArray(object === null || object === void 0 ? void 0 : object.messageIds) ? object.messageIds.map((e) => String(e)) : [],
        };
    },
    toJSON(message) {
        const obj = {};
        message.topicName !== undefined && (obj.topicName = message.topicName);
        if (message.messageIds) {
            obj.messageIds = message.messageIds.map((e) => e);
        }
        else {
            obj.messageIds = [];
        }
        return obj;
    },
    create(base) {
        return AwsSdkSnsTags.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseAwsSdkSnsTags();
        message.topicName = (_a = object.topicName) !== null && _a !== void 0 ? _a : undefined;
        message.messageIds = ((_b = object.messageIds) === null || _b === void 0 ? void 0 : _b.map((e) => e)) || [];
        return message;
    },
};
if (_m0__default["default"].util.Long !== Long__default["default"]) {
    _m0__default["default"].util.Long = Long__default["default"];
    _m0__default["default"].configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}

exports.AwsApiGatewayTags = AwsApiGatewayTags;
exports.AwsApiGatewayTags_AwsApiGatewayRequestTags = AwsApiGatewayTags_AwsApiGatewayRequestTags;
exports.AwsLambdaInitializationTags = AwsLambdaInitializationTags;
exports.AwsLambdaInvocationTags = AwsLambdaInvocationTags;
exports.AwsLambdaRuntimeTags = AwsLambdaRuntimeTags;
exports.AwsLambdaTags = AwsLambdaTags;
exports.AwsSdkDynamodbTags = AwsSdkDynamodbTags;
exports.AwsSdkSnsTags = AwsSdkSnsTags;
exports.AwsSdkSqsTags = AwsSdkSqsTags;
exports.AwsSdkTags = AwsSdkTags;
exports.AwsSnsEventTags = AwsSnsEventTags;
exports.AwsSqsEventTags = AwsSqsEventTags;
exports.AwsTags = AwsTags;
exports.HttpRouterTags = HttpRouterTags;
exports.awsLambdaTags_OutcomeFromJSON = awsLambdaTags_OutcomeFromJSON;
exports.awsLambdaTags_OutcomeToJSON = awsLambdaTags_OutcomeToJSON;
exports.awsLambdaTags_ResponseModeFromJSON = awsLambdaTags_ResponseModeFromJSON;
exports.awsLambdaTags_ResponseModeToJSON = awsLambdaTags_ResponseModeToJSON;
exports.protobufPackage = protobufPackage;
