// @serverless/sdk-schema/dist/tags/notice.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Long = require('long');
var _m0 = require('protobufjs/minimal');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var Long__default = /*#__PURE__*/_interopDefaultLegacy(Long);
var _m0__default = /*#__PURE__*/_interopDefaultLegacy(_m0);

/* eslint-disable */
const protobufPackage = "serverless.instrumentation.tags.v1";
exports.NoticeTags_NoticeType = void 0;
(function (NoticeTags_NoticeType) {
    /** NOTICE_TYPE_UNSPECIFIED - Notice type not set */
    NoticeTags_NoticeType[NoticeTags_NoticeType["NOTICE_TYPE_UNSPECIFIED"] = 0] = "NOTICE_TYPE_UNSPECIFIED";
    /**
     * NOTICE_TYPE_SDK_INTERNAL - Signals SDK noteworthy event
     * Event that's not relevant to the user but can be important instruction for the Console
     * on how to process monitoring data of given case
     */
    NoticeTags_NoticeType[NoticeTags_NoticeType["NOTICE_TYPE_SDK_INTERNAL"] = 1] = "NOTICE_TYPE_SDK_INTERNAL";
    NoticeTags_NoticeType[NoticeTags_NoticeType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(exports.NoticeTags_NoticeType || (exports.NoticeTags_NoticeType = {}));
function noticeTags_NoticeTypeFromJSON(object) {
    switch (object) {
        case 0:
        case "NOTICE_TYPE_UNSPECIFIED":
            return exports.NoticeTags_NoticeType.NOTICE_TYPE_UNSPECIFIED;
        case 1:
        case "NOTICE_TYPE_SDK_INTERNAL":
            return exports.NoticeTags_NoticeType.NOTICE_TYPE_SDK_INTERNAL;
        case -1:
        case "UNRECOGNIZED":
        default:
            return exports.NoticeTags_NoticeType.UNRECOGNIZED;
    }
}
function noticeTags_NoticeTypeToJSON(object) {
    switch (object) {
        case exports.NoticeTags_NoticeType.NOTICE_TYPE_UNSPECIFIED:
            return "NOTICE_TYPE_UNSPECIFIED";
        case exports.NoticeTags_NoticeType.NOTICE_TYPE_SDK_INTERNAL:
            return "NOTICE_TYPE_SDK_INTERNAL";
        case exports.NoticeTags_NoticeType.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
function createBaseNoticeTags() {
    return { message: "", type: undefined };
}
const NoticeTags = {
    encode(message, writer = _m0__default["default"].Writer.create()) {
        if (message.message !== "") {
            writer.uint32(10).string(message.message);
        }
        if (message.type !== undefined) {
            writer.uint32(16).int32(message.type);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0__default["default"].Reader ? input : _m0__default["default"].Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseNoticeTags();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.message = reader.string();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.type = reader.int32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            message: isSet(object.message) ? String(object.message) : "",
            type: isSet(object.type) ? noticeTags_NoticeTypeFromJSON(object.type) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        message.message !== undefined && (obj.message = message.message);
        message.type !== undefined &&
            (obj.type = message.type !== undefined ? noticeTags_NoticeTypeToJSON(message.type) : undefined);
        return obj;
    },
    create(base) {
        return NoticeTags.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b;
        const message = createBaseNoticeTags();
        message.message = (_a = object.message) !== null && _a !== void 0 ? _a : "";
        message.type = (_b = object.type) !== null && _b !== void 0 ? _b : undefined;
        return message;
    },
};
if (_m0__default["default"].util.Long !== Long__default["default"]) {
    _m0__default["default"].util.Long = Long__default["default"];
    _m0__default["default"].configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}

exports.NoticeTags = NoticeTags;
exports.noticeTags_NoticeTypeFromJSON = noticeTags_NoticeTypeFromJSON;
exports.noticeTags_NoticeTypeToJSON = noticeTags_NoticeTypeToJSON;
exports.protobufPackage = protobufPackage;
