// @serverless/sdk-schema/dist/tags/warning.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Long = require('long');
var _m0 = require('protobufjs/minimal');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var Long__default = /*#__PURE__*/_interopDefaultLegacy(Long);
var _m0__default = /*#__PURE__*/_interopDefaultLegacy(_m0);

/* eslint-disable */
const protobufPackage = "serverless.instrumentation.tags.v1";
exports.WarningTags_WarningType = void 0;
(function (WarningTags_WarningType) {
    /** WARNING_TYPE_UNSPECIFIED - Warning type not set */
    WarningTags_WarningType[WarningTags_WarningType["WARNING_TYPE_UNSPECIFIED"] = 0] = "WARNING_TYPE_UNSPECIFIED";
    /** WARNING_TYPE_USER - Warning explicitly reported by user */
    WarningTags_WarningType[WarningTags_WarningType["WARNING_TYPE_USER"] = 1] = "WARNING_TYPE_USER";
    /** WARNING_TYPE_SDK_USER - Warning reported internally by the SDK that signal potential misusage on user side */
    WarningTags_WarningType[WarningTags_WarningType["WARNING_TYPE_SDK_USER"] = 2] = "WARNING_TYPE_SDK_USER";
    /** WARNING_TYPE_SDK_INTERNAL - Warning reported internally by the SDK that signal non fatal SDK issue */
    WarningTags_WarningType[WarningTags_WarningType["WARNING_TYPE_SDK_INTERNAL"] = 3] = "WARNING_TYPE_SDK_INTERNAL";
    WarningTags_WarningType[WarningTags_WarningType["UNRECOGNIZED"] = -1] = "UNRECOGNIZED";
})(exports.WarningTags_WarningType || (exports.WarningTags_WarningType = {}));
function warningTags_WarningTypeFromJSON(object) {
    switch (object) {
        case 0:
        case "WARNING_TYPE_UNSPECIFIED":
            return exports.WarningTags_WarningType.WARNING_TYPE_UNSPECIFIED;
        case 1:
        case "WARNING_TYPE_USER":
            return exports.WarningTags_WarningType.WARNING_TYPE_USER;
        case 2:
        case "WARNING_TYPE_SDK_USER":
            return exports.WarningTags_WarningType.WARNING_TYPE_SDK_USER;
        case 3:
        case "WARNING_TYPE_SDK_INTERNAL":
            return exports.WarningTags_WarningType.WARNING_TYPE_SDK_INTERNAL;
        case -1:
        case "UNRECOGNIZED":
        default:
            return exports.WarningTags_WarningType.UNRECOGNIZED;
    }
}
function warningTags_WarningTypeToJSON(object) {
    switch (object) {
        case exports.WarningTags_WarningType.WARNING_TYPE_UNSPECIFIED:
            return "WARNING_TYPE_UNSPECIFIED";
        case exports.WarningTags_WarningType.WARNING_TYPE_USER:
            return "WARNING_TYPE_USER";
        case exports.WarningTags_WarningType.WARNING_TYPE_SDK_USER:
            return "WARNING_TYPE_SDK_USER";
        case exports.WarningTags_WarningType.WARNING_TYPE_SDK_INTERNAL:
            return "WARNING_TYPE_SDK_INTERNAL";
        case exports.WarningTags_WarningType.UNRECOGNIZED:
        default:
            return "UNRECOGNIZED";
    }
}
function createBaseWarningTags() {
    return { message: "", stacktrace: undefined, type: undefined };
}
const WarningTags = {
    encode(message, writer = _m0__default["default"].Writer.create()) {
        if (message.message !== "") {
            writer.uint32(10).string(message.message);
        }
        if (message.stacktrace !== undefined) {
            writer.uint32(26).string(message.stacktrace);
        }
        if (message.type !== undefined) {
            writer.uint32(16).int32(message.type);
        }
        return writer;
    },
    decode(input, length) {
        const reader = input instanceof _m0__default["default"].Reader ? input : _m0__default["default"].Reader.create(input);
        let end = length === undefined ? reader.len : reader.pos + length;
        const message = createBaseWarningTags();
        while (reader.pos < end) {
            const tag = reader.uint32();
            switch (tag >>> 3) {
                case 1:
                    if (tag !== 10) {
                        break;
                    }
                    message.message = reader.string();
                    continue;
                case 3:
                    if (tag !== 26) {
                        break;
                    }
                    message.stacktrace = reader.string();
                    continue;
                case 2:
                    if (tag !== 16) {
                        break;
                    }
                    message.type = reader.int32();
                    continue;
            }
            if ((tag & 7) === 4 || tag === 0) {
                break;
            }
            reader.skipType(tag & 7);
        }
        return message;
    },
    fromJSON(object) {
        return {
            message: isSet(object.message) ? String(object.message) : "",
            stacktrace: isSet(object.stacktrace) ? String(object.stacktrace) : undefined,
            type: isSet(object.type) ? warningTags_WarningTypeFromJSON(object.type) : undefined,
        };
    },
    toJSON(message) {
        const obj = {};
        message.message !== undefined && (obj.message = message.message);
        message.stacktrace !== undefined && (obj.stacktrace = message.stacktrace);
        message.type !== undefined &&
            (obj.type = message.type !== undefined ? warningTags_WarningTypeToJSON(message.type) : undefined);
        return obj;
    },
    create(base) {
        return WarningTags.fromPartial(base !== null && base !== void 0 ? base : {});
    },
    fromPartial(object) {
        var _a, _b, _c;
        const message = createBaseWarningTags();
        message.message = (_a = object.message) !== null && _a !== void 0 ? _a : "";
        message.stacktrace = (_b = object.stacktrace) !== null && _b !== void 0 ? _b : undefined;
        message.type = (_c = object.type) !== null && _c !== void 0 ? _c : undefined;
        return message;
    },
};
if (_m0__default["default"].util.Long !== Long__default["default"]) {
    _m0__default["default"].util.Long = Long__default["default"];
    _m0__default["default"].configure();
}
function isSet(value) {
    return value !== null && value !== undefined;
}

exports.WarningTags = WarningTags;
exports.protobufPackage = protobufPackage;
exports.warningTags_WarningTypeFromJSON = warningTags_WarningTypeFromJSON;
exports.warningTags_WarningTypeToJSON = warningTags_WarningTypeToJSON;
